llvm/lib/Target/WebAssembly/WebAssemblyCFGStackify.cpp

```diff
@@ -1943,8 +1951,16 @@ bool WebAssemblyCFGStackify::fixCallUnwindMismatches(MachineFunction &MF) {
   // When end_loop is before end_try_table within the same BB in unwind
   // destinations, we should split the end_loop into another BB.
   if (!WebAssembly::WasmUseLegacyEH)
-    for (auto &[UnwindDest, _] : UnwindDestToTryRanges)
-      splitEndLoopBB(UnwindDest);
+    for (auto &[UnwindDest, _] : UnwindDestToTryRanges) {
+      auto It = EHPadToTry.find(UnwindDest);
+      // If UnwindDest is the faker caller block, it will not be in EHPadToTry
+      // map
+      if (It != EHPadToTry.end()) {
+        auto *TryTable = It->second;
+        auto *EndTryTable = BeginToEnd[TryTable];
+        splitEndLoopBB(EndTryTable->getParent());
+      }
+    }
```

Comment:
This changed because after splitting the BB in `addUnreachableAfterTryTables` to add an `unreachable`, `end_try_table` is not in the EHPad anymore.
- Before:
```
bb:
  ... try body ...

ehpad:
  end_try_table
  end_block
  ... catch handler ...
```

- After:
```
bb:
  ... try body ...

end_try_table_bb:
  end_try_table
  unreachable

ehpad:
  end_block
  ... catch handler ...
```

Line 2198~2206 change in `fixCatchUnwindMismatches` is the same thing. `splitEndLoopBB`'s argument name changed from `UnwindDest` to `EndTryTableBB` for the same reason.